documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler-implement.adoc

```diff
@@ -156,6 +156,60 @@ It is therefore mandatory to use `CompletableFuture` APIs that are invoked even
 Failing to do so may result in the `Handler` `callback` parameter to never be completed, causing the request processing to hang forever.
 ====
 
+[[pg-server-http-handler-impl-request-listeners]]
+====== `Request` Listeners
+
+Application may add listeners to the `Request` object to be notified of particular events happening during the request/response processing.
+
+`Request.addIdleTimeoutListener(Predicate<TimeoutException>)` allows you to add an idle timeout listener, which is invoked when an idle timeout period elapses during the request/response processing, if the idle timeout event is not notified otherwise.
+
+The idle timeout listener is invoked only when:
+
+* There is no pending demand for request content; that is, there is not an outstanding call to `Request.demand(Runnable)`.
+* There is no pending response write; that is, there is not an outstanding call to `Response.write(boolean, ByteBuffer, Callback)`.
```

Comment:
Is this new behaviour?  I thought the point of the idle callback was the boolean, so that it could be ignored in some situations....   or are all the situations in which we ignore it when we don't have a read or a write pending?